keps/0002-kong-gateway-api.md

```diff
@@ -260,17 +312,61 @@ In order to consider this KEP `implemented` we must have the following present:
 
 [docs]:https://docs.konghq.com/kubernetes-ingress-controller/
 
+### Graduation Criteria
+
+The following highlights milestones along the path of Gateway support maturity.
+
+Individual milestones will be marked `provisional` or `implementable` independently, in accordance with whether the design details have enough planning and specification available yet to support them, and complete milestones will be marked as `implemented`:
+
+- `provisional` - don't start work until the previous milestones have been completed **and** the [design details](#design-details) for related components and features have been completed.
+- `implementable` - this is supported by the [design details](#design-details) and is ready for work to start
+- `implemented` - the related components and functionality are already available now at the designated quality level
+
+The milestones may correlate directly with [Github Milestones][github-milestones] in the same repository.
+
+[github-milestones]:https://github.com/Kong/kubernetes-ingress-controller/milestones
+
+#### Milestone 1 - Alpha Quality - Initial HTTP Support (implementable)
+
+- [ ] our validating webhook for Gateway resources has been added and provides errors for unimplemented features
+- [ ] a `Gateway` controller implementation with basic support for "existing" operational mode is introduced
+- [ ] an initial implementation of `HTTPRoute` is added which includes support for the majority of options
+- [ ] integration tests added which cover all the supported features of `HTTPRoute`
+
+#### Milestone 2 - Alpha Quality - Extended API Support (provisional)
+
+- [ ] an initial implementation of `TCPRoute` is added which includes support for the majority of features
+- [ ] integration tests added which cover all the features of `TCPRoute`
+- [ ] an initial implementation of `UDPRoute` is added which includes support for the majority of features
+- [ ] integration tests added which cover all the features of `UDPRoute`
+- [ ] an initial implementation of `TLSRoute` is added which includes support for the majority of features
+- [ ] integration tests added which cover all the features of `TLSRoute`
+
+#### Milestone 3 - Alpha Quality - Operator Support (provisional)
+
+- [ ] operator support for provisioning and managing the lifecycle of `Gateway` resources is added
+- [ ] operator has support for automatic upgrades of provisioned `Gateways`
+- [ ] operator has support for health checking and rollback mechanisms for provisioned `Gateways`
+- [ ] operator has prometheus metrics which capture information about `Gateway` operations, including error alerts for failures e.t.c.
+- [ ] integration tests added which cover all features of `Gateway`
+
+#### Milestone 4 - Beta Quality (provisional)
```

Comment:
The `alpha`/`beta` wording as used in this document may be confusing, as it may interfere with the `alpha`/`beta` suffixes we use for KIC versioning. Would "feature preview" be a better word than "alpha"/"beta"?

&lt;/bikeshedding>

Reply:
So I had some anticipation of this and this ties directly into this discussion: https://github.com/Kong/team-k8s/discussions/154 about feature gates. I'm thinking in terms of feature gate maturity levels. If we ultimately decide as a team that we're not in favor of feature gates then yeah maybe a different wording makes sense, like preview.

Reply:
Discussed over zoom: Kong/team-k8s#154 is an open discussion pondering the two known approaches to enabling/disabling GW API support in KIC, and the decision on that matter is purely technical and doesn't need to block this KEP.